### PR TITLE
Merge 3.x into Master

### PR DESCRIPTION
**Pull request by dmaicher**

### CODE REVIEW

File: src/Builder/AbstractFormContractor.php [hunk 1]
```diff
@@ -150,6 +157,18 @@ private function isAnyInstanceOf(?string $type, array $classes): bool
             }
         }
 
+        // handle form type inheritance and check all parent types
+        $resolvedType = $this->formRegistry->getType($type);
+        if (null !== $resolvedType->getParent()) {
```
Comment (VincentLanglet) [inline src/Builder/AbstractFormContractor.php:162]: You have to create a variable 
$parent = $resolvedType->getParent();
if (null !== $parent)

Comment (dmaicher) [reply src/Builder/AbstractFormContractor.php:162]: yeah will check it later

File: tests/Builder/AbstractFormContractorTest.php [hunk 2]
```diff
@@ -97,6 +111,7 @@ public function testDefaultOptionsForSonataFormTypes(): void
             ModelListType::class,
             ModelHiddenType::class,
             ModelAutocompleteType::class,
+            'MyCustomType',
```
Comment (VincentLanglet) [inline tests/Builder/AbstractFormContractorTest.php:114]: Easiest way to solve this, would be to create a fixture `MyCustomType` class.

Comment (dmaicher) [reply tests/Builder/AbstractFormContractorTest.php:114]: Maybe `stdClass::class` would also work

Comment (dmaicher) [reply tests/Builder/AbstractFormContractorTest.php:114]: but cleaner and easier to understand is indeed a `MyCustomType` fixtures class

File: src/Builder/AbstractFormContractor.php [hunk 3]
```diff
@@ -159,8 +159,8 @@ private function isAnyInstanceOf(?string $type, array $classes): bool
 
         // handle form type inheritance and check all parent types
         $resolvedType = $this->formRegistry->getType($type);
-        if (null !== $resolvedType->getParent()) {
-            $parentType = \get_class($resolvedType->getParent()->getInnerType());
+        if (null !== $parentType = $resolvedType->getParent()) {
```
Comment (VincentLanglet) [inline src/Builder/AbstractFormContractor.php:162]: Personnally, I prefer to avoid defining variable in a if, but seems like no linter is against this.

**Review by VincentLanglet — APPROVED** [review]

File: src/Builder/AbstractFormContractor.php (hunk 3 above)
Comment (dmaicher) [reply src/Builder/AbstractFormContractor.php:162]: I don't mind changing it :blush: Shall I?

**Comment by dmaicher** [comment]
@VincentLanglet considering that quite a few people are using the master branch (for Symfony 5 compat): shall we wait with merging this breaking change? Until the persistence bundles are fixed (https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/pull/607 + https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1473) this will break things for people updating to the latest dev commit

**Comment by VincentLanglet** [comment]
> @VincentLanglet considering that quite a few people are using the master branch (for Symfony 5 compat): shall we wait with merging this breaking change? Until the persistence bundles are fixed ([sonata-project/SonataDoctrineMongoDBAdminBundle#607](https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/pull/607) + [sonata-project/SonataDoctrineORMAdminBundle#1473](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1473)) this will break things for people updating to the latest dev commit

Sure we can wait a little.

**Comment by dmaicher** [comment]
@VincentLanglet now we are good to go :blush: persistence bundles are compatible

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**